rclpy/test/test_destruction.py

```diff
@@ -214,14 +214,10 @@ def test_destroy_service_asap():
             with service.handle:
                 pass
 
-            with service.handle:
-                node.destroy_service(service)
-                # handle valid because it's still being used
-                with service.handle:
-                    pass
+            node.destroy_service(service)
```

Comment:
I know this was done for the `Client` class, but I am curios why it was removed. Isn't this a valid test? /cc @sloretz

Reply:
It was an intentional behavior change. When someone requests that a client be destroyed, `Destroyable` will prevent new things from using it so that it can be destructed when the current uses of it end.

Reply:
Ah, I see. Thanks for the explanation.